frontend/src/components/ChatInput.tsx

```diff
@@ -25,6 +24,7 @@ function ChatInput({ disabled, onSendMessage }: ChatInputProps) {
   };
 
   const onKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (disabled) return;
     if (event.key === "Enter" && !event.shiftKey && !isComposing) {
       event.preventDefault(); // prevent a new line
       handleSendChatMessage();
```

Comment:
```suggestion
    if (event.key === "Enter" && !event.shiftKey && !isComposing) {
      event.preventDefault(); // prevent a new line
      if (!disabled) {
          handleSendChatMessage();
      }
```

Currently, we prevent `Enter` from adding a newline if composing or the shift key is pressed. IMO this should happen regardless of whether the input is disabled, for consistency.